### ISSUE TITLE
2 subcharts with different versions results in one not being rendered during dry-run/install

### ISSUE THREAD

**Issue by tldoty** [helm/helm#4310]
Output of `helm version`:
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}

Output of `kubectl version`:
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:21:50Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3+coreos.0", GitCommit:"f588569ed1bd4a6c986205dd0d7b04da4ab1a3b6", GitTreeState:"clean", BuildDate:"2018-02-10T01:42:55Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
AWS

I have a Chart with a requirements.yaml that looks like the following:

```
dependencies:  
  - name: chartA
    repository: "LOCAL-REPO"
    version: 1.0.0
  - name: chartB
    repository: "LOCAL-REPO"
    version: 2.0.0
    alias: "friendlyNameB"
  - name: chartB
    repository: "LOCAL-REPO"
    version: 1.0.0
    alias: "friendlyNameC"
```

I deploy chartB twice with different override configurations and also different versions of chartB.
In the output of my dry-run only one of the chartB templates get rendered. When they both use the same version then all templates render as expected. Is it expected to use the same version of subcharts of the same origin?

**Comment by patoarvizu** [comment]
I'm seeing similar behavior but sounds like it's under different conditions. In my case I'm trying to deploy different aliases of the same version of a chart (with different values) and it still only deploys one set. I tried deploying different versions and I'm still seeing the same issue.

In my case, it looks like `helm dep up` just ignores the `alias` field, or at least it doesn't save it in `requirements.lock` and under `charts/` I only see one chart, as opposed to one chart per alias, as implied by this statement in the documentation: "The manual way of achieving this is by copy/pasting the same chart in the charts/ directory multiple times with different names." (https://docs.helm.sh/developing_charts/#alias-field-in-requirements-yaml). Is this happening in your case too?

**Comment by jascott1** [comment]
Hi @tldoty This is probably "undefined behavior" but I suspect that 'alias' implies same chart and same version. As @patoarvizu points out, the docs essentially say "copy/pasting the _same_ chart".

@patoarvizu IIRC you are correct and `helm dep` is not alias aware. You should manually control the aliased charts on disk.

**Comment by patoarvizu** [comment]
@jascott1 Wait, I don't get it. Then what's the point of having `alias`es if I'll have to manually control the versions? Are you saying that `alias` is just the mechanism to have multiple versions of the same chart? That would seem to contradict the example given here: https://docs.helm.sh/developing_charts/#alias-field-in-requirements-yaml

**Comment by jascott1** [comment]
@patoarvizu Alias was designed for multiple installs of the same chart (including the version). Sorry for the delay in reply. Did you find a suitable solution?

**Comment by patoarvizu** [comment]
We believe this is an issue with the version of Helm we're currently using (2.6.1). The original problem was addressed with a different approach, although we ran into it again later for a different problem and we ended up doing some stitching together of things via pure gomplate and bash. Basically a very raw version of kustomize.

If there are no plans to patch 2.6 (or if this is not believed to be an issue), this ticket can be closed.
